packages/client-sdk-nodejs/src/batchutils/batch-functions.ts

```diff
@@ -0,0 +1,145 @@
+import {CacheDelete, CacheGet, CacheSet} from '@gomomento/sdk-core';
+import {CacheClient} from '../cache-client';
+import {
+  BatchDeleteRequest,
+  BatchDeleteResponse,
+  BatchGetRequest,
+  BatchGetResponse,
+  BatchSetItem,
+  BatchSetRequest,
+  BatchSetResponse,
+  defaultMaxConcurrentRequests,
+} from './batch-props';
+import {range} from '@gomomento/sdk-core/dist/src/internal/utils';
+
+export {
+  BatchDeleteRequest,
+  BatchDeleteResponse,
+  BatchGetRequest,
+  BatchGetResponse,
+  BatchSetItem,
+  BatchSetRequest,
+  BatchSetResponse,
+  defaultMaxConcurrentRequests,
+} from './batch-props';
+
+export async function batchGet(
+  request: BatchGetRequest
+): Promise<BatchGetResponse> {
+  const maxConcurrentGets = request.maxConcurrentGets
+    ? request.maxConcurrentGets
+    : Math.min(defaultMaxConcurrentRequests, request.keys.length);
+
+  const batchGetResults = range(maxConcurrentGets).map((workerId: number) =>
+    getWorker(workerId, request.cacheClient, request.cacheName, request.keys)
+  );
+  const awaitAll = await Promise.all(batchGetResults);
+
+  const batchGetResponse: BatchGetResponse = {};
+  awaitAll.forEach(responses => {
+    for (const key of Object.keys(responses)) {
+      batchGetResponse[key] = responses[key];
+    }
+  });
+  return batchGetResponse;
+}
+
+async function getWorker(
+  workerId: number,
+  cacheClient: CacheClient,
+  cacheName: string,
+  keys: Array<string>
```

Comment:
you probably already know this, and I also might have mentioned it in the ticket, but this only works in TS/JS because the node runtime is single threaded. Otherwise it would be very important to use a thread-safe data structure here since multiple workers are sharing a reference to it.